beekeeper-metadata-cleanup/src/main/java/com/expediagroup/beekeeper/metadata/cleanup/handler/ExpiredMetadataHandler.java

```diff
@@ -58,6 +61,31 @@ public ExpiredMetadataHandler(CleanerClientFactory cleanerClientFactory,
     return housekeepingMetadataRepository.findRecordsForCleanupByModifiedTimestamp(instant, pageable);
   }
 
+  @Override
+  public void disableTable(String database, String table) {
+    housekeepingMetadataRepository.deleteScheduledOrFailedPartitionRecordsForTable(database, table);
+    HousekeepingMetadata tableRecord = housekeepingMetadataRepository.findTableRecord(database, table);
+    tableRecord.setHousekeepingStatus(DISABLED);
+    housekeepingMetadataRepository.save(tableRecord);
+  }
+
+  @Override
+  public boolean tableHasBeekeeperProperty(HousekeepingMetadata metadata) {
+    try (CleanerClient client = cleanerClientFactory.newInstance()) {
+      Map<String, String> properties = client.getTableProperties(metadata.getDatabaseName(),
+          metadata.getTableName());
+      if (properties != null) {
+        String beekeeperProperty = properties.get(UNREFERENCED.getTableParameterName());
+        return beekeeperProperty != null && beekeeperProperty.equals("true");
```

Comment:
`"true".equals(beekeeperProperty)` is NPE safe

Reply:
ah, nice